test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -470,6 +675,15 @@ public void testObjLitRedefinedInGlobalScope() {
   public void testObjLitRedefinedInLocalScope() {
     test("var a = {}; a.b = {c: 0}; function d() { a.b = {c: 1}; } e(a.b.c);",
          "var a$b = {c: 0}; function d() { a$b = {c: 1}; } e(a$b.c);");
+
+    test("var a = {};/** @nocollapse */ a.b = {c: 0}; "
+        + "function d() { a.b = {c: 1}; } e(a.b.c);",
+        "var a = {}; a.b = {c: 0}; function d() { a.b = {c: 1}; } e(a.b.c);");
```

Comment:
testSame?

In general, all tests without a $ in the output code should probably be testSame.